### PR TITLE
feat: add no resources page (#1337)

### PR DESCRIPTION
**Pull request by lstocchi**
### What does this PR do?

This PR displays a panel with a message when there are no resources in the resources page

@mairin i downloaded the icon from penpot. As it looks similar to a premium icon from fontawesome (https://fontawesome.com/icons/engine?f=classic&s=solid) can you confirm i can use the one I downloaded? Thank you!!

This is a first iteration and will be followed by https://github.com/containers/podman-desktop/issues/1934

### Screenshot/screencast of this PR

image

### What issues does this PR fix or reference?

it is part of #1337 

### How to test this PR?

1. be sure to not have any resources
2.  go to the resources page

1b. edit the code to make the message appears when there are providers.

### CODE REVIEW

File: packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte [hunk 1]
```diff
@@ -174,124 +175,137 @@ async function startContainerProvider(
...
-                </Tooltip>
-              {/if}
-            </div>
+    {#if providers.length === 0}
+      <div class="p-2 flex flex-col bg-zinc-800 rounded-lg mt-4">
+        <div class="flex flex-col items-center text-center mt-10 mb-20">
+          <EngineIcon />
+          <p class="text-lg text-gray-200">No resource found</p>
+          <div class="text-sm text-gray-400 mt-2">
+            <p>Create a container engine provider or install an extension to start</p>
```
Comment (benoitf) [inline packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte]: It's not limited to container engine, can be kubernetes engine as well
Also extension need to be started not only installed

```suggestion
            <p>Start an extension that manage container or Kubernetes engines</p>
```

Comment (benoitf) [reply packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte]: Sentence is just a suggestion, can be anything else that includes 'Kubernetes engine' as well

**Comment by mairin** [comment]
@lstocchi I drew that one by hand, but I didnt realize it looked so closs to the font awesome one. Let me tweak it so it's a bit different, I will post a new one here.

**Comment by mairin** [comment]
Here you go :) This is what it looks like in situ:
Screenshot from 2023-04-04 09-58-32

Here's the SVG:
containerengine_pathsv2

File: packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte [hunk 2]
```diff
@@ -174,124 +175,137 @@ async function startContainerProvider(
...
-              <div class="flex mt-1">
-                <ConnectionStatus status="{kubeConnection.status}" />
-              </div>
-              {#if kubeConnection.status === 'started'}
-                <div class="mt-2">
-                  <div class="text-gray-400 text-xs">Kubernetes endpoint</div>
-                  <div class="mt-1">
-                    <span class="my-auto text-xs">{kubeConnection.endpoint.apiURL}</span>
+          </div>
+          <!-- podman machines columns -->
```
Comment (benoitf) [inline packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte]:
```suggestion
          <!-- connection columns -->
```
need to remove 'podman' there

File: packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte [hunk 3]
```diff
@@ -174,124 +175,137 @@ async function startContainerProvider(
...
-                </Tooltip>
-              {/if}
-            </div>
+    {#if providers.length === 0}
+      <div class="p-2 flex flex-col bg-zinc-800 rounded-lg mt-4">
+        <div aria-label="no-resource-panel" class="flex flex-col items-center text-center mt-10 mb-20">
+          <EngineIcon />
+          <p class="text-lg text-gray-200">No resource found</p>
+          <div class="text-sm text-gray-400 mt-2">
+            <p>Start an extension that manage container or Kubernetes engines</p>
```
Comment (benoitf) [inline packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte]: Can we use `EmptyScreen` component from @deboer-tim to display Empty state ?

Comment (lstocchi) [reply packages/renderer/src/lib/preferences/PreferencesResourcesRendering.svelte]: Good idea, done 👍

**Review by benoitf — APPROVED** [review]

**Pull request merged**